src/QMCWaveFunctions/LCAO/MultiQuinticSpline1D.h

```diff
@@ -223,6 +222,92 @@ class MultiQuinticSpline1D
     }
   }
 
+
+  inline void batched_evaluateVGL_multiCenter(const OffloadArray2D& r,
+                                              OffloadArray4D& vgl,
+                                              T Rmax,
+                                              size_t num_centers) const
+  {
+    const size_t nElec           = r.size(0);
+    const size_t Nxyz_per_center = r.size(1) / num_centers; // Assuming r.size(1) = Nxyz * num_centers
+    assert(3 == vgl.size(0));
+    assert(nElec == vgl.size(1));
+    assert(r.size(1) == vgl.size(2));
+    const size_t nRnl = vgl.size(3);
+
+    const size_t total_positions = nElec * Nxyz_per_center * num_centers;
+
+    double one_over_log_delta = myGrid.OneOverLogDelta;
+    T lower_bound             = myGrid.lower_bound;
+    T dlog_ratio              = myGrid.LogDelta;
+
+    auto* r_ptr   = r.device_data();
+    auto* u_ptr   = vgl.device_data_at(0, 0, 0, 0);
+    auto* du_ptr  = vgl.device_data_at(1, 0, 0, 0);
+    auto* d2u_ptr = vgl.device_data_at(2, 0, 0, 0);
+
+    auto* coeff_ptr       = coeffs_.device_data();
+    auto* first_deriv_ptr = first_deriv_.device_data();
+    const size_t nCols    = coeffs_.cols();
+    const size_t coefsize = coeffs_.size();
+    const auto nsplines   = num_splines_;
+
+    PRAGMA_OFFLOAD("omp target teams distribute parallel for collapse(3) \
+                    is_device_ptr(first_deriv_ptr, coeff_ptr, r_ptr, u_ptr, du_ptr, d2u_ptr)")
+    for (size_t i_c = 0; i_c < num_centers; ++i_c)
```

Comment:
This looks like the same as the batched_evaluateVGL (i.e. loops over nelec/npbc/ncenter can be flattened)